content/posts/building-images-with-img.md

```diff
@@ -0,0 +1,255 @@
+As mentioned in the [`img` GitHub README](https://github.com/genuinetools/img#running-with-kubernetes), setting `securityContext.procMount` to `Unmasked` is no longer required to run `img`, but it does enable PID namespace isolation. Essentially, this prevents child processes from executing `kill -2` against the parent `img` process. This `procMount` option was introduced by a series of patches (ex. [this pull request](https://github.com/kubernetes/kubernetes/pull/64283)) against Kubernetes and other projects.
+
+So why is that option commented out in our `Pod` above? The answer lies in current Kubernetes support, both for the `securityContext.procMount` option and the `allowedProcMountTypes` specification in our `PodSecurityPolicy`. While `procMount` has been [enabled by default since Kubernetes 1.12](https://github.com/genuinetools/img#running-with-kubernetes), `allowedProcMountTypes` is [still an Alpha feature](https://kubernetes.io/docs/reference/command-line-tools-reference/feature-gates/#feature-gates-for-alpha-or-beta-features).
+
+This leaves us with a few options to handle the `procMount` specification, depending on the nature of our Kubernetes cluster:
+
+1. If running a cluster that allows admin access to the Kubernetes master, we can selectively enable the Feature Gate for `allowedProcMountTypes` in our `PodSecurityPolicy`. Because GKE doesn't allow this access, this isn't an option in this example.
+2. We can [create an Alpha GKE cluster](https://cloud.google.com/kubernetes-engine/docs/concepts/alpha-clusters) which will enable the field in question, among many others. Because of the  functional limitations of Alpha clusters, this is only an option for sandbox workloads.
+3. Because the required field is `PodSecurityPolicy`-specific, we could disable these policies across the entire cluster. However, this goes against security best practices.
+4. Finally, we could simply omit the `securityContext.procMount: Unmasked` option in our `Pod` definition, which disables PID namespace isolation. We'll use this option for the remainder of the blog post.
+
+## Using `img` in a Jenkins Pipeline
+With our Kubernetes and Google Cloud resources now configured, we'll now look at running our `img` `Pod` from a Jenkins Pipeline. 
+
+To execute this Jenkins job, we'll set up a [Multibranch Pipeline job](https://jenkins.io/doc/book/pipeline/multibranch/) that pulls from our [`img-pipeline` GitHub repository](https://github.com/mdelgin/img-pipeline). Here is the `Jenkinsfile` we'll use:
+```groovy
+pipeline {
+  agent {
+    kubernetes {
+      label "img"
+      yamlFile 'img-resources/imgPod.yaml'
+    }
+  }
+  stages {
+    stage('Build and Push') {
+      steps {
+        container('img') {
+          sh """
+            img build -t gcr.io/melgin/img-hello-world ./hello-world
+            /home/user/google-cloud-sdk/bin/gcloud auth print-access-token | img login -u oauth2accesstoken --password-stdin https://gcr.io
```

Comment:
Based on recommended change for img image suggest changing to the following:
`gcloud  auth configure-docker --quiet`